pyuvdata/tests/test_miriad.py

```diff
@@ -153,19 +154,23 @@ def test_wronglatlon():
     lonfile = os.path.join(DATA_PATH, 'zen.2456865.60537_wronglon.xy.uvcRREAA')
     telescopefile = os.path.join(DATA_PATH, 'zen.2456865.60537_wrongtelecope.xy.uvcRREAA')
 
-    uvtest.checkWarnings(uv_in.read_miriad, [latfile], nwarnings=2,
-                         message=['Altitude is not present in file and latitude value does not match',
-                                  latfile + ' was written with an old version of pyuvdata'])
-    uvtest.checkWarnings(uv_in.read_miriad, [lonfile], nwarnings=2,
-                         message=['Altitude is not present in file and longitude value does not match',
-                                  lonfile + ' was written with an old version of pyuvdata'])
+    uvtest.checkWarnings(uv_in.read_miriad, [latfile], nwarnings=3,
+                         message=['Altitude is not present in file and latitude value does not match value',
+                                  'This file was written with an old version of pyuvdata',
+                                  'This file was written with an old version of pyuvdata'])
+    uvtest.checkWarnings(uv_in.read_miriad, [lonfile], nwarnings=3,
+                         message=['Altitude is not present in file and longitude value does not match value '
+                                  'for PAPER in known telescopes. Using values from known telescopes.',
+                                  'This file was written with an old version of pyuvdata',
+                                  'This file was written with an old version of pyuvdata'])
     uvtest.checkWarnings(uv_in.read_miriad, [telescopefile], {'run_check': False},
-                         nwarnings=4,
+                         nwarnings=5,
                          message=['Altitude is not present in Miriad file, and telescope',
                                   'Telescope location is set at sealevel at the '
                                   'file lat/lon coordinates. Antenna positions '
                                   'are present, but the mean antenna position',
-                                  telescopefile + ' was written with an old version of pyuvdata',
+                                  'This file was written with an old version of pyuvdata',
+                                  'This file was written with an old version of pyuvdata',
```

Comment:
why are all these warnings doubled?

Reply:
Ah, its because I now call `_load_antpos` twice in a `read_miriad` call: once at the beginning when I'm loading in the metadata, and once at the end when all of the heavy-weight metadata have been loaded in. The first call to `_load_antpos` isn't needed, but its done anyways b/c its lumped in with the `read_miriad_metadata` call (its very fast anyways so speed isn't a concern). But it means that any warning that the function raises is technically raised twice in a `read_miriad` call. We can change this if this is too annoying..

Reply:
It's not too big a deal. Could just the first one suffice (it makes sense to get it when you only read the metadata), or do you really need the later one?

Reply:
Good question. Not entirely sure, but it looks like the `Nants_telescope` is set using `sorted_unique_ants`, the latter of which is constructed from the `ant_1_array` and `ant_2_arrray` (the heavy-weight metadata arrays). The second call to `_load_antpos` is fed this `sorted_unique_ants`, so it seems like the second call to `_load_antpos` has some added functionality that the first call cannot reproduce...

Reply:
Ok, so here's what's going on there. Miriad doesn't understand the concept of antenna numbers -- it essentially requires that the index of the antenna position is the number of that antenna. The problem with that is that sometimes you don't have all the antennas present in a file (like for HERA where we haven't built some of them yet). You could re-number all the antennas to be consecutive, but then a given physical antenna could have a different number for different files which would drive people analyzing the data crazy. So instead we pad the antenna positions with zeros for the indices that do not correspond to existing antenna information. 

It is perfectly reasonable to have antenna position information for antennas that you don't have any data for -- that's why there's a difference between Nants_telescope & Nants_data.  So to calculate Nants_telescope we could just use the non-zero positions, except that some files seem to be missing antenna positions for antennas that *do* appear in the data. So to calculate Nants_telescope we use a union of the antennas that have non-zero positions and the antennas that are present in the data.

So if you just read the antpos variable without reading the data, you can't necessary down-select it to get just the valid antenna positions because you don't know if there are antennas in the data for which the antenna positions are set to zero for some reason.

So now that I understand the complexity, I think you're doing the right thing and we should just live with duplicated warnings (this is assuming that you have to read the data to know what antennas are present in the data, which I believe is true for miriad).

Reply:
Yeah, I agree with that summary.